plugins/e57/io/E57Writer.cpp

```diff
@@ -81,12 +82,25 @@ void E57Writer::ChunkWriter::write(pdal::PointRef& pt)
     }
 
     // Add point to buffer and increase index
+    using DimId = pdal::Dimension::Id;
     for (auto& keyValue: m_doubleBuffers)
     {
         auto pdaldim = pdal::e57plugin::e57ToPdal(keyValue.first);
-        if (pdaldim != pdal::Dimension::Id::Unknown)
+        if (pdaldim != DimId::Unknown)
         {
             auto val = pt.getFieldAs<double>(pdaldim);
+            if ((pdaldim == DimId::Red || pdaldim == DimId::Green ||
+                    pdaldim == DimId::Blue) &&
+                    val > colorLimit)
+            {
+                colorLimit = colorLimit << 8;  // Increase color bytes.
```

Comment:
I think that we can infer the color and intensity limit based on the data type used for storing this dimension. Could you please check if that is feasible?

Reply:
I tried this.
Dimension datatype is always `Unsigned16` even if colors are in 0->255 range.

Reply:
This would mean that the reader is wrong then. Could you please double check what's happening here? Maybe it would make sense to ask on PDAL repo?